x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/preview_rules_route.ts

```diff
@@ -173,26 +187,62 @@ export const previewRulesRoute = async (
           }
         };
 
-        const signalRuleAlertType = signalRulesAlertType({
-          ...ruleOptions,
-          lists: context.lists,
-          config,
-          indexNameOverride: previewIndex,
-          ruleExecutionLogClientOverride: previewRuleExecutionLogClient,
-          // unused as we override the ruleExecutionLogClient
-          eventLogService: {} as unknown as IEventLogService,
-          eventsTelemetry: undefined,
-          ml: undefined,
-          refreshOverride: 'wait_for',
-        });
-
-        await runExecutors(
-          signalRuleAlertType.executor,
-          signalRuleAlertType.id,
-          signalRuleAlertType.name,
-          previewRuleParams,
-          alertInstanceFactoryStub
-        );
+        switch (previewRuleParams.type) {
+          case 'query':
+            const queryAlertType = previewRuleTypeWrapper(createQueryAlertType(ruleOptions));
+            await runExecutors(
+              queryAlertType.executor,
+              queryAlertType.id,
+              queryAlertType.name,
+              previewRuleParams,
+              alertInstanceFactoryStub
+            );
+            break;
+          case 'threshold':
+            const thresholdAlertType = previewRuleTypeWrapper(
+              createThresholdAlertType(ruleOptions)
+            );
+            await runExecutors(
+              thresholdAlertType.executor,
+              thresholdAlertType.id,
+              thresholdAlertType.name,
+              previewRuleParams,
+              alertInstanceFactoryStub
+            );
+            break;
+          case 'threat_match':
+            const threatMatchAlertType = previewRuleTypeWrapper(
+              createIndicatorMatchAlertType(ruleOptions)
+            );
+            await runExecutors(
+              threatMatchAlertType.executor,
+              threatMatchAlertType.id,
+              threatMatchAlertType.name,
+              previewRuleParams,
+              alertInstanceFactoryStub
+            );
+            break;
+          case 'eql':
+            const eqlAlertType = previewRuleTypeWrapper(createEqlAlertType(ruleOptions));
+            await runExecutors(
+              eqlAlertType.executor,
+              eqlAlertType.id,
+              eqlAlertType.name,
+              previewRuleParams,
+              alertInstanceFactoryStub
+            );
+            break;
+          case 'machine_learning':
```

Comment:
That was a technical limitation before and with our updates, I actually think we'd be able to since we use the real executors. It's not planned for this release but we should be able to implement it soon

Reply:
Do you recall the technical limitation? I do not, but my guess is that it had to do with the various states ML jobs could be in, and how to reflect those in the preview; the rule executor itself isn't really more complex than a normal query rule.

We can test this and see how it behaves, and similar to Indicator Match rules we can decide to turn this preview type off before release.

Reply:
I added it in the latest commit, there's no performance issues as far as I can tell, I'd actually expect this rule type to be one of the quicker executions just due to the typical use case. The errors were mostly already handled by the existing ML error handling we have in the executors and we bubble it all up with no problems. Let me know what you think but in my opinion I feel pretty confident in adding this given we're using the actual executors now